packages/odyssey-react-mui/src/useAutocomplete.tsx

```diff
@@ -0,0 +1,398 @@
+    const rowRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+      if (rowRef.current) {
+        /**
+         * Checking for child height to workaround a bug where the clientHeight of the row isn't updated correctly
+         * @see here if you need to know more: https://github.com/bvaughn/react-window/issues/582#issuecomment-1883074908
+         */
+        const firstChild = rowRef.current.firstElementChild;
+        const height = firstChild
+          ? firstChild.clientHeight
+          : rowRef.current.clientHeight;
+
+        setItemSize(height);
+      }
+    }, [index, rowRef, setItemSize]);
+
+    const baseOption = data[index];
+    const { key, props } = baseOption;
+
+    /**
+     * react-window calculates the absolute positions of the list items, via an inline style, so
+     * we need to add it to each list item that is being rendered in the viewable list window.
+     * @see here if you need to know more: https://github.com/bvaughn/react-window?tab=readme-ov-file#why-is-my-list-blank-when-i-scroll
+     */
+    const styles = useMemo(
+      () => ({
+        ...style,
+        height: "auto",
+      }),
+      [style],
+    );
+
+    return (
+      <div ref={rowRef}>
+        <li {...props} key={key} style={styles} />
+      </div>
+    );
+  };
+
+  const OuterListboxContext = createContext({});
+
+  const OuterListboxElementType = forwardRef<HTMLDivElement>((props, ref) => {
+    const outerProps = useContext(OuterListboxContext);
+    return <div ref={ref} {...props} {...outerProps} />;
+  });
+
+  const useResetCache = (length: number) => {
+    const resetCacheRef = useRef<VariableSizeList>(null);
+    useEffect(() => {
+      if (resetCacheRef.current) {
+        resetCacheRef.current.resetAfterIndex(0, true);
+      }
+    }, [length]);
+    return resetCacheRef;
+  };
+
+  const ListboxComponent = forwardRef<
+    HTMLDivElement,
+    HTMLAttributes<HTMLElement>
+  >(function (props, ref) {
+    const [listHeight, setListHeight] = useState(0);
+
+    const { children, ...other } = props;
+    const itemData: ReactElement[] = (children as ReactElement[]).flatMap(
+      (item: ReactElement & { children?: ReactElement[] }) =>
+        [item].concat(item.children || []),
+    );
+
+    const sizeMap = useRef<number[]>([]);
+
+    const getListBoxHeight = useCallback(() => {
+      // 8px of padding top/bottom applied by MUI
+      const COMBINED_LISTBOX_PADDING = 16;
+
+      if (itemData.length > OVERSCAN_ROW_COUNT) {
+        // has a max-height of 40vh set in CSS. This is only set because height needs to be a number
+        return 99999;
+      } else {
+        const itemsHeightCalculated = itemData
+          .map((_, index) => sizeMap.current[index] || 0)
+          .reduce(
+            (prevItemHeight, nextItemHeight) => prevItemHeight + nextItemHeight,
+            0,
+          );
+        return COMBINED_LISTBOX_PADDING + itemsHeightCalculated;
+      }
```

Comment:
This code is confusing because it looks like we're looking `itemData`, but in reality, all we care about are the item indexes. Since they're sequential, we can create a new array with that many items.

```suggestion
        const itemsHeightCalculated = Array(itemData.length).fill(null)
          .map((_, index) => sizeMap.current[index] || 0)
          .reduce(
            (prevItemHeight, nextItemHeight) => prevItemHeight + nextItemHeight,
            0,
          );
        return COMBINED_LISTBOX_PADDING + itemsHeightCalculated;
      }
```

But even better, we can use `slice`:

```suggestion
        const itemsHeightCalculated = sizeMap.current
          .slice(0, itemData.length - 1)
          .reduce(
            (prevItemHeight, nextItemHeight = 0) => prevItemHeight + nextItemHeight,
            0,
          );
        return COMBINED_LISTBOX_PADDING + itemsHeightCalculated;
      }
```

---

Also, if `sizeMap` is a ref, we should call it `sizeMapRef`. Having a random `.current` out there is confusing. My assumption was that `sizeMap` is a `new Map()`.